pkg/infradb/infradb.go

```diff
@@ -257,6 +257,29 @@ func UpdateLB(lb *LogicalBridge) error {
 
 	return nil
 }
+func removeVniFromVpns(vni uint32) error {
+	vpns := make(map[uint32]bool)
+	if vni != 0 {
+		found, err := infradb.client.Get("vpns", &vpns)
+		if err != nil {
+			log.Println(err)
+			return err
+		}
+		if !found {
+			log.Println("removeVniFromVpns(): No VPNs have been found")
+			log.Println(vni, ":")
+			return ErrKeyNotFound
+		}
+		delete(vpns, vni)
+
+		err = infradb.client.Set("vpns", &vpns)
+		if err != nil {
+			log.Println(err)
+			return err
```

Comment:
again, here we print an error msg and return that err.
On the level above, we print the same error. What value do we add by behaving that way? Probably we just flood the log with the identical error messages.
Each error should be handled once, and logging is a form of handling. You should either log it or return it, not both